orchestra/actions/util/impl.py

```diff
@@ -33,18 +60,37 @@ def _run_script(
     :param stderr: passed as the "stderr" parameter to subprocess.run
     :return: a subprocess.CompletedProcess instance
     """
-    if strict_flags:
-        script_to_run = bash_prelude
-    else:
-        script_to_run = ""
 
-    if environment:
-        script_to_run += export_environment(environment)
+    script_to_run = _wrap_script(script, environment, strict_flags, cwd)
+    logger.log(loglevel, f"The following script is going to be executed:\n{script.strip()}\n")
+    return subprocess.run(["/bin/bash", "-c", script_to_run], stdout=stdout, stderr=stderr)
 
-    script_to_run += script
 
-    logger.log(loglevel, f"The following script is going to be executed:\n" + script.strip())
-    return subprocess.run(["/bin/bash", "-c", script_to_run], stdout=stdout, stderr=stderr, cwd=cwd)
+def _exec_script(
+    script,
+    environment: Optional[Mapping] = None,
+    strict_flags=True,
+    cwd=None,
+    loglevel="INFO",
+) -> NoReturn:
+    """Helper for exec-ing into a shell scripts.
+    :param script: the script to run
+    :param environment: will be exported at the beginning of the script
+    :param strict_flags: if True, a prelude is prepended to the script to help catch errors
+    :param cwd: if not None, the command is executed in the specified path
+    :param loglevel: log debug informations at this level
+    """
+
+    script_to_run = _wrap_script(script, environment, strict_flags, cwd)
+    logger.log(loglevel, f"The following script is going to be exec'ed into:\n{script.strip()}\n")
+
+    # Fork to allow the deletion of temporary directories/files by the child
+    pid = os.fork()
+    if pid == 0:
+        sys.exit(0)
```

Comment:
```
$ python3.10 -m pytest test/commands/test_shell.py::test_shell^
...
        # Fork to allow the deletion of temporary directories/files by the child
        pid = os.fork()
        if pid == 0:
>           sys.exit(0)
E           SystemExit: 0

orchestra/actions/util/impl.py:90: SystemExit
----------------------------------------------------------------------------------- Captured stderr setup ------------------------------------------------------------------------------------
warning: templates not found in /dev/null
Cloning into '/tmp/pytest-of-***/pytest-16/test_shell0/orchestra'...
warning: templates not found in /dev/null
done.
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
[+] WARNING - The remote HEADs cache does not exist, you should run `orchestra update`
================================================================================== short test summary info ===================================================================================
FAILED test/commands/test_shell.py::test_shell - SystemExit: 0
===================================================================================== 1 failed in 0.13s ======================================================================================
```

Reply:
Found decent way to handle the problem, fix pushed